### ISSUE TITLE
Question on the pooled_out design

### ISSUE THREAD

**Issue by sysu-zjw** [facebookresearch/DPR#107]
Hi. I find the `pooled_output` was defined as the first position of `sequence_output` at [here](https://github.com/facebookresearch/DPR/blob/42161470d6f16d20c20f6ea2516941c224fc0b89/dpr/models/hf_models.py#L189). In my understanding they are different, and rename one as another may be confusing. Just for curiosity, I wonder if this is a typo or is there any theory/experiment to show that `sequence_output[:, 0, :]` is more reasonable than the `pooled_output`? Thanks.

**Comment by vlad-karpukhin** [comment]
Hi @sysu-zjw ,
yes the naming might not be good for that.
I did try transformer's pooled_output out signal at first - it didn't work well for me somehow. But that was quite a long time ago before I even tried in-batch negatives.
Overall, adding some additional learnable layers wasn't working well for me (I tried a bunch of different options including pooled_output, extra projection, aggregating signals from multiple transformer layers, etc.)
You can still try to use  pooled_output in the current codebase. Would be interesting to know if it works now.

**Comment by sysu-zjw** [comment]
@vlad-karpukhin Thanks for your reply. I try things similar and also find their benefit is not significant.

**Comment by sysu-zjw** [comment]
I just try using the `pooled_output` on my dataset. To my surprise, the result is much worse than `sequence_output[:, 0, :]` (in-batch acc 50% v.s. 90%). This is a bit weird.

By the way, may I also confirm the in-batch acc detail with you? I am using 1 `positive_ctx` and 1 `hard_negative_ctx` for each query, `dev_batch_size` is 16, training on 8 gpu device. So, does the `correct prediction ratio` mean picking the positive one out of 16+16=32 candidate? (I got this question as I never see my in-batch acc below 50% no matter what kind of mistake I made during modifying this framework)
